GVFS/GVFS.Common/Git/GitSsl.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+using GVFS.Common.Tracing;
+
+namespace GVFS.Common.Git
+{
+    public class GitSsl

```

Comment:
Add unit tests for this class?

Reply:
I'd love to, but it's a bit challenging, due to the fact, that certificate store has no interface, that we could mock and I don't think we want to write certificates to the actual store during unit testing, as this might not work on CI servers. Also, the certificates are being validated using native libraries (that is - it's outside our control), so it would be very hard to setup a valid certificate. If you have ideas how to approach this - they would be appreciated.

Reply:
Those are great points.  @kewillford, @jrbriggs, do you have any thoughts on additional tests that could be added to validate `GitSsl`?

My main concern is that we might inadvertently break this feature\functionality if we don't have sufficient coverage of it in our unit and functional tests.

Reply:
Absolutely, covering this with tests is something, that I think all of us want. One way to achieve this is to add additional layers of abstraction, like:
```
interface ICertificateLoader
{
   X509Certificate2 LoadCertificateBySubjectName(string subjectName);
}

interface ICertificateVerifier
{
   bool VerifyCertificate(X509Certificate2 certificate);
}
```
Then use them in our GitSsl class and mock those in tests. That's something, that would work, but I'd like to hear your opinion on the solution first, as I didn't see similar code in GVFS code base.

Reply:
So, finally got time to get back to this and pushed unit tests for GitSsl. Would you have a look @wilbaker ?